File: SAMPLES/PERL/hello.pl
#!/usr/bin/perl
# hello.pl - Exemple PERL86 : print, scalaires, concatenation
# Compile : PERL86.EXE SAMPLES\PERL\hello.pl hello.asm

use strict;
use warnings;

my $name = "PERL86";
my $vers = "1.0";
my $msg  = "Compilateur Perl -> 8086";

print "Bonjour de " . $name . " v" . $vers . "\n";
print $msg . "\n";

# Arithmetique simple
my $a = 10;
my $b = 32;
my $c = $a + $b;
print "10 + 32 = ";
print $c;
print "\n";

# Concatenation avec .
my $s1 = "Hello";
my $s2 = " World";
my $s3 = $s1 . $s2;
print $s3;
print "\n";

# Concatenation composee .=
my $full = "Perl";
$full .= " 86";
$full .= " rocks";
print $full;
print "\n";

# Repetition x
my $line = "-" x 20;
print $line;
print "\n";

File: SAMPLES/PYTHON/hello.py
# hello.py - Exemple PYTHON86 : print, variables, chaines, entiers
# Compile : PYTHON86.EXE SAMPLES\PYTHON\hello.py hello.asm

# Affichage simple
print("Bonjour de PYTHON86 !")
print("Compilateur Python -> 8086")
print()

# Variables entieres
x = 42
y = 8
print("x =", x)
print("y =", y)

# Arithmetique
z = x + y
print("x + y =", z)
z = x - y
print("x - y =", z)
z = x * y
print("x * y =", z)

# Variables chaines
nom = "PYTHON86"
version = "1.0"
print("Nom :", nom)
print("Version :", version)

# Concatenation de chaines
msg = "Hello" + " " + "World"
print(msg)

# Repetition de chaines
sep = "=-" * 20
print(sep)

# Entiers negatifs
n = -7
print("n =", n)

# Expressions complexes
resultat = (x + y) * 2 - 10
print("(x + y) * 2 - 10 =", resultat)

print()
print("Fin de hello.py")
